staking/types/validator.go

```diff
@@ -27,10 +28,10 @@ var (
 
 // ValidatorWrapper contains validator and its delegation information
 type ValidatorWrapper struct {
-	Validator           `json:"validator" yaml:"validator"`
-	Delegations         []Delegation `json:"delegations" yaml:"delegations"`
-	SnapshotValidator   *Validator   `json:"snapshot_validator" yaml:"snaphost_validator"`
-	SnapshotDelegations []Delegation `json:"snapshot_delegations" yaml:"snapshot_delegations"`
+	Validator           `json:"validator" yaml:"validator" rlp:"nil"`
+	Delegations         []Delegation `json:"delegations" yaml:"delegations" rlp:"nil"`
+	SnapshotValidator   *Validator   `json:"snapshot_validator" yaml:"snaphost_validator" rlp:"nil"`
```

Comment:
only the *Validator can be nil right?    Others don't have to have "nil" tag. See example:

AccountNonce hexutil.Uint64  `json:"nonce"      gencodec:"required"`
		Price        *hexutil.Big    `json:"gasPrice"   gencodec:"required"`
		GasLimit     hexutil.Uint64  `json:"gas"        gencodec:"required"`
		ShardID      uint32          `json:"shardID"    gencodec:"required"`
		ToShardID    uint32          `json:"toShardID"`
		Recipient    *common.Address `json:"to"         rlp:"nil"`
		Amount       *hexutil.Big    `json:"value"      gencodec:"required"`
		Payload      hexutil.Bytes   `json:"input"      gencodec:"required"`
		V            *hexutil.Big    `json:"v" gencodec:"required"`
		R            *hexutil.Big    `json:"r" gencodec:"required"`
		S            *hexutil.Big    `json:"s" gencodec:"required"`
		Hash         *common.Hash    `json:"hash" rlp:"-"`

Reply:
Delegation can be nil as well if no body delegate. I saw similar decoding issue as well. Also, SnapshotValidator is not implemented and hence is nil. Thus, to be safe, I added nil to all places. Later, we can check one by one, until we are sure it's not nil.